src/sync/sync_session.cpp

```diff
@@ -259,23 +259,39 @@ const SyncSession::State& SyncSession::State::active = Active();
 const SyncSession::State& SyncSession::State::dying = Dying();
 const SyncSession::State& SyncSession::State::inactive = Inactive();
 
-SyncSession::SyncSession(SyncClient& client, std::string realm_path, SyncConfig config, bool force_client_resync)
-: m_handle_refresh([this](util::Optional<app::AppError> error) {
-    if (error) {
-        // 10 seconds is arbitrary
-        std::this_thread::sleep_for(std::chrono::milliseconds(10000));
-        if (m_state == &State::active) {
-            user()->refresh_custom_data(m_handle_refresh);
+std::function<void(util::Optional<app::AppError>)> _impl::handle_refresh(std::shared_ptr<SyncSession> session) {
+    return [session](util::Optional<app::AppError> error) {
+        if (!session)
+            return;
+
+        auto session_user = session->user();
+        auto is_user_expired = session_user &&
+            session_user->refresh_jwt().expires_at < std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now().time_since_epoch()).count();
```

Comment:
Anything using `std::chrono` really wants a `using namespace std::chrono` within the function.